src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingSubmissionService.java

```diff
@@ -194,31 +202,50 @@ private void updateGitDiffReport(ProgrammingExerciseParticipation programmingExe
      * If this is the case, the submission is set to {@link SubmissionType#ILLEGAL}.
      *
      * @param programmingExerciseParticipation current participation of the exam exercise
      * @param programmingSubmission            new created submission of the repository commit
      */
-    private void checkForIllegalExamSubmission(ProgrammingExerciseParticipation programmingExerciseParticipation, ProgrammingSubmission programmingSubmission) {
+    private void checkForIllegalSubmission(ProgrammingExerciseParticipation programmingExerciseParticipation, ProgrammingSubmission programmingSubmission) {
         ProgrammingExercise programmingExercise = programmingExerciseParticipation.getProgrammingExercise();
-        boolean isExamExercise = programmingExercise.isExamExercise();
-        // Students are not allowed to submit a programming exercise after the exam due date, if this happens we set the Submission to ILLEGAL
-        if (isExamExercise && programmingExerciseParticipation instanceof ProgrammingExerciseStudentParticipation) {
-            var optionalStudent = ((ProgrammingExerciseStudentParticipation) programmingExerciseParticipation).getStudent();
-            Optional<User> optionalStudentWithGroups = optionalStudent.isPresent() ? userRepository.findOneWithGroupsAndAuthoritiesByLogin(optionalStudent.get().getLogin())
-                    : Optional.empty();
-            if (optionalStudentWithGroups.isPresent() && !examSubmissionService.isAllowedToSubmitDuringExam(programmingExercise, optionalStudentWithGroups.get(), true)) {
-                final String message = "The student " + optionalStudentWithGroups.get().getLogin()
-                        + " just illegally submitted code after the allowed individual due date (including the grace period) in the participation "
-                        + programmingExerciseParticipation.getId() + " for the exam programming exercise " + programmingExercise.getId();
+        // Students are not allowed to submit a programming exercise after the due date, if this happens we set the Submission to ILLEGAL
+        if (programmingExerciseParticipation instanceof ProgrammingExerciseStudentParticipation studentParticipation) {
+            var optionalStudent = studentParticipation.getStudent();
+            var optionalStudentWithGroups = optionalStudent.flatMap(student -> userRepository.findOneWithGroupsAndAuthoritiesByLogin(student.getLogin()));
+            if (optionalStudentWithGroups.isEmpty()) {
+                return;
+            }
+            User student = optionalStudentWithGroups.get();
+            if (!isAllowedToSubmit(studentParticipation, student, programmingSubmission)) {
+                final String message = "The student %s just illegally submitted code after the allowed individual due date (including the grace period) in the participation %d for the programming exercise %d"
```

Comment:
```suggestion
                final String message = "The student %s illegally submitted code after the allowed individual due date (including the grace period) in the participation %d for the programming exercise %d"
```
(Not sure about this, but) I would think that the "just" aspect is already clear by the timestamp on the log